src/components/Map/Map.tsx

```diff
@@ -82,14 +87,31 @@ export default function Map({ countries, disputedAreas, fcsData, alertData }: Ma
           setCountryData,
           setCountryIso3Data,
           setRegionNutritionData,
-          setIpcRegionData
-        );
-        setSelectedCountryName(selectedCountryData.properties.adm0_name);
+          setIpcRegionData,
+          setIsDataAvailable
+        ).then(() => {
+          setIsLoading(false);
+        });
```

Comment:
We don't use then(), instead await the fetchCountryData promise and after that setIsLoading to false